### ISSUE TITLE
Should we have a :todo named argument in test routines?

### ISSUE THREAD

**Issue by JJ** [Raku/problem-solving#188]
There is a proposed named argument for test routines like `ok` or `is`:

* `:todo` would allow it to fail with an explanation why.

A PR to provide for it (partially) was done [here](https://github.com/rakudo/rakudo/pull/368) and updated [here](https://github.com/rakudo/rakudo/pull/3665). But `:todo` is not part of the S24 speculation, although there are skipped tests written for it here:

https://github.com/Raku/roast/blob/master/S24-testing/1-basic.t#L27-L31

So do we need this? Do we want this?

**Comment by AlexDaniel** [comment]
Changed the ticket a bit, we already have `:desc` implemented everywhere, it's just the `:todo` that is in question here.

**Comment by AlexDaniel** [comment]
Some related commits:
* (2009-06-25) https://github.com/perl6/mu/commit/d6da37b6f7 [ere-smo] added &is to Test.pm
* (2006-07-09) https://github.com/perl6/mu/commit/2687c30443 added t/02-test-pm/ which consolidates tests for …
* (2005-08-11) https://github.com/perl6/mu/commit/71db8dd7e4 ...undo...
* (2005-08-11) https://github.com/perl6/mu/commit/26e52f822b Initial Pugs checkin.
* (2005-04-22) https://github.com/perl6/mu/commit/cb75ba229e start of the removal of todo_* functions; see ext/Test/ChangeLog for details;

**Comment by vrurg** [comment]
What actual problem `:todo` solves? The thing I know for sure is its use would complicate fudging of spectests and most likely we'd have to explicitly disable it in specs.

**Comment by jnthn** [comment]
I don't think a `:todo` option adds a great deal of value over the existing form. I guess one might argue that it's easier to delete the todo marker together with the test, but it's probably more common that you keep the test and delete the todo. And in that case, the ergonomics are not better with a `:todo` named argument, since in most editors it's easier to delete a line holding a call to `todo` as a whole than it is to delete an argument.

Even if one can argue there's some value, then it has to overcome the costs. We'd need to implement it on every test assertion routine. It'd need documenting. It'd be two ways to do the same thing, which ain't automatically bad, but there's usually one obvious right way to do things, and it's hard to see that in this case.

In short, it doesn't seem to pull its weight. If we were starting a test library from scratch it'd be an interesting API design discussion, perhaps. But that's not where we are, and it's far from clear we'd want to go in the direction of `:todo` as an adverb on a test.
